### DISCUSSION TITLE
Set table and field description

### DISCUSSION THREAD

**Discussion by elenaaralla** [fluentmigrator/fluentmigrator#1663] [General] [answered]
Hi,

is there a way to set table description and filed description when create table? (see attached image)
I need this for database documentation.

Thank you,
Elena.
![Schermata 2023-06-22 alle 16 54 51](https://github.com/fluentmigrator/fluentmigrator/assets/14870650/af6239a3-1e20-4e4e-bd07-f3a990342650)

**Comment by gfgw** [comment]
As far as I know (but I am just a simple user of FluentMigrator) you can only do this with
`Alter.Table("tableName").AddColumn("columnName").WithColumnDescription("description")`

I have an existing table with an existing column, and I would like to change the description. Unfortunately that also seems impossible.

**Reply by elenaaralla** [reply, marked as answer]
Hi @gfgw,

it's possible to add description also on table creation; see belowe code:

```
  public class DefaultDB_20230627_151500_CreatePecMailbox : AutoReversingMigration
    {
        public override void Up()
        {
            Create.Table("PecMailboxes")
                .WithDescription("Tabella con anagrafica delle caselle PEC.")
                .InSchema("pdl")
                .WithColumn("Id")
                    .AsInt32()
                    .Identity()
                    .PrimaryKey()
                    .NotNullable()
                    .WithColumnDescription("Chiave primaria - identificativo della casella PEC.")
                .WithColumn("Email")
                    .AsString(200)
                    .NotNullable()
                    .WithColumnDescription("Codice dell'errore.")
```

thank you,
Elena.

**Reply by gfgw** [reply]
Thank you Elena.

Thanks to your answer I just discovered that you can only use .WithColumnDescription after specifying the column type.
This does not work:
`this.Alter.Table("tableName").AlterColumn("columnName").WithColumnDescription("description")`
But this does work:
`this.Alter.Table("tableName").AlterColumn("columnName").AsInt32().WithColumnDescription("description")`

This solves my problem.
